lib/features/mailbox_dashboard/presentation/action/dashboard_action.dart

```diff
@@ -163,4 +163,22 @@ class SearchEmailByFromFieldsAction extends DashBoardAction {
 
 class CloseSearchEmailViewAction extends DashBoardAction {}
 
-class CancelSelectionSearchEmailAction extends DashBoardAction {}
\ No newline at end of file
+class CancelSelectionSearchEmailAction extends DashBoardAction {}
+
+class OpenAdvancedSearchViewAction extends DashBoardAction {
+  final BuildContext context;
+
+  OpenAdvancedSearchViewAction(this.context);
+
+  @override
+  List<Object?> get props => [context];
+}
+
+class ClearSearchFilterAppliedAction extends DashBoardAction {
+  final BuildContext context;
+
+  ClearSearchFilterAppliedAction(this.context);
+
+  @override
+  List<Object?> get props => [context];
+}
```

Comment:
IMO, it doesn't matter too much in this case.
1. Because it's part of the same processing screen.
2. We use BuildContext quite a lot in test conditions, not just AppLocalizations

Reply:
Removed all `BuildContext` in actions. `BuildContext` should only be used in widgets.